### DISCUSSION TITLE
dispatching to nav2 targets on the command line....

### DISCUSSION THREAD

**Discussion by jerrynsr** [iRobotEducation/create3_docs#69] [Questions] [answered]
Is there a method  to record a waypoint within Rviz2 or other to a file ?  I would like to wrote a python application that will  dispatch the robot to a room closest to an outdoor Blink Camera event,  barking along the way, linger around the room a bit and then return to my hobby room and recharge until the next event.    Events are going to be time and webhook driven.   A typical timed even would be to roll into my home office while I'm working... and make dog breathing sounds... sleep there.    At random times leave and roam the house.
Additional events might be triggered by voice recognition.   I don't want to use Google.. I'm going to use an open source solution running locally on the RPI4.    Once I have a fixed focus Oak D Lite camera installed I will trigger events based on facial recognition.
Some people it will like and make happy dog sounds,  other for others it will growl.

I'm happy with the progress I made so far.    Right now if an object has a pedestal to bot detects it, but since it's too close when the bot rotates left/right to navigate around the object it hazard sensor detects it's blocked and stops.. led on top flashes yellow.

Thanks everyone for the help, I appreciate it.

By the way.. all the graphical ui applications such as Rviz2 runs remotely on my PC.  rmw_fastrtps_cpp works extremely well.

**Comment by justinIRBT** [comment, marked as answer]
Regarding recording a waypoint, `ros2 bag record` will subscribe to topics, but that might not be what you are looking for (more documentation here is you are curious https://docs.ros.org/en/galactic/Tutorials/Ros2bag/Recording-And-Playing-Back-Data.html).  If you just want to see the position of a waypoint that you can then put in your python script to execute in a time sequence, you could just use `ros2 topic echo` on the waypoint topic to print out what the coordinates are, then code them into your python application.

Regarding the flashing yellow light, did you modify the `safety_override` mode?  If you are using the default mode `none`, the robot has limited ability to back up because it doesn't have rear cliffs.  If you don't have any places where the robot could back off a cliff, you could change the `safety_override` mode to `backup_only` or `full` depending on the speed you want, see more detail here:
https://iroboteducation.github.io/create3_docs/api/safety/
If you want to keep safety on, you may be able to edit your Nav2 config to lower your negative x Cartesian velocity limits so that the Nav2 stack won't try to back up as much, although its native algorithms may not perform well with that limitation when it gets into tight situations.

**Reply by justinIRBT** [reply]
Also note that the `safety_override` param can be set in the Application configuration page on the robot's web server if you want to permanently set its value.  It is already set to `none` there by default, so you just need to change the value there, save the config on the webserver, then restart the application to take effect
